api/preprints/serializers.py

```diff
@@ -469,72 +451,46 @@ def require_admin_permission():
             description = validated_data['description']
             self.set_field(preprint.set_description, description, auth)
-            save_preprint = True
-
-        if validated_data.get('article_doi'):
-            article_doi = validated_data['article_doi']
-            # use different regex from what we use in validate_doi as we should find and set full correct doi string.
-            # it should start with "10." and have at least one group of any characters with a not required slash
-            # e.g. 10.12/, 10.1234/test, 10.798797789/test/, 10.1234/test1/test2/random
-            # e.g.2. from http://doi.com/10.32/test/random we fetch 10.32/test/random
-            stripped_article_doi = re.search(r'10\.\w+\/([\w\-\.]*\/?)+', article_doi)
-            if not stripped_article_doi:
-                raise exceptions.ValidationError('The `article_doi` format is incorrect')
-
-            stripped_article_doi = stripped_article_doi.group()
-            doi = settings.DOI_FORMAT.format(prefix=preprint.provider.doi_prefix, guid=preprint._id)
-            if doi.startswith(stripped_article_doi):
-                raise exceptions.ValidationError(
-                    detail=f'The `article_doi` "{doi}" is already associated with this'
-                           f' preprint please enter a peer-reviewed publication\'s DOI',
-                )
 
-            preprint.article_doi = stripped_article_doi
-            save_preprint = True
-        else:
-            preprint.article_doi = None
-            save_preprint = True
+        preprint.article_doi = validated_data.get('article_doi')
 
         if 'license_type' in validated_data or 'license' in validated_data:
             license_details = get_license_details(preprint, validated_data)
             self.set_field(preprint.set_preprint_license, license_details, auth)
-            save_preprint = True
 
         if 'original_publication_date' in validated_data:
             preprint.original_publication_date = validated_data['original_publication_date'] or None
-            save_preprint = True
 
         if 'custom_publication_citation' in validated_data:
             preprint.custom_publication_citation = validated_data['custom_publication_citation'] or None
-            save_preprint = True
 
         if published is not None:
             if not preprint.primary_file:
                 raise exceptions.ValidationError(
                     detail='A valid primary_file must be set before publishing a preprint.',
                 )
             self.set_field(preprint.set_published, published, auth)
-            save_preprint = True
             recently_published = published
             preprint.set_privacy('public', log=False, save=True)
 
-        if save_preprint:
-            preprint.save()
+        try:
+            preprint.full_clean()
+        except ValidationError as e:
+            raise exceptions.ValidationError(detail=str(e))
+
+        preprint.save()
```

Comment:
It's okay to save here if it's a no-op, Django is aware of that as long as this in the request context. It won't make a db transaction if there's no change.